x-pack/plugins/security_solution/public/management/pages/trusted_apps/state/async_data_binding.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Immutable } from '../../../../../common/endpoint/types';
+
+export interface UninitialisedAsyncBinding {
+  type: 'UninitialisedAsyncBinding';
+}
+
+export interface InProgressAsyncBinding<T, E> {
+  type: 'InProgressAsyncBinding';
+  previousBinding: StaleAsyncBinding<T, E>;
+}
+
+export interface PresentAsyncBinding<T> {
```

Comment:
Similar to my comment below, I think `CurrentAsyncBinding` may be more readable

Reply:
All types were renamed.